File: components/content_creation/notes/core/note_service.cc
// Copyright 2021 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "components/content_creation/notes/core/note_service.h"

#include "base/functional/callback.h"
#include "components/content_creation/notes/core/note_features.h"

namespace content_creation {

NoteService::NoteService(std::unique_ptr<TemplateStore> template_store)
    : template_store_(std::move(template_store)) {}

NoteService::~NoteService() = default;

void NoteService::GetTemplates(GetTemplatesCallback callback) {
  DCHECK(IsStylizeEnabled());
  template_store_->GetTemplates(std::move(callback));
}

}  // namespace content_creation

File: components/content_creation/notes/android/java/src/org/chromium/components/content_creation/notes/NoteService.java
// Copyright 2021 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.components.content_creation.notes;

import org.chromium.base.Callback;
import org.chromium.components.content_creation.notes.models.NoteTemplate;

import java.util.List;

/**
 * Interface for interacting with the native note service. The service is
 * responsible for fetching templates.
 */
public interface NoteService {
    /**
     * Called to get the list of available Note templates.
     */
    void getTemplates(Callback<List<NoteTemplate>> callback);
}
